src/conversion/conversion.js

```diff
@@ -102,6 +114,347 @@ export default class Conversion {
+	 *					style: {
+	 *						'font-weight': 'bold'
+	 *					}
+	 *				},
+	 *				viewElement => {
+	 *					const fontWeight = viewElement.getStyle( 'font-weight' );
+	 *
+	 *					if ( viewElement.is( 'span' ) && fontWeight && /\d+/.test() && Number( fontWeight ) > 500 ) {
+	 *						return {
+	 *							name: true,
+	 *							style: [ 'font-weight' ]
+	 *						};
+	 *					}
+	 *				}
+	 *			]
+	 *		} );
+	 *
+	 *		// Conversion from/to a model attribute key which value is an enum (`fontSize=big|small`).
+	 *		// `upcastAlso` set as callback enables a conversion of a wide range of different view elements.
+	 *		conversion.attributeToElement( {
+	 *			model: {
+	 *				key: 'fontSize',
+	 *				values: [ 'big', 'small' ]
+	 *			},
+	 *			view: {
+	 *				big: {
+	 *					name: 'span',
+	 *					style: {
+	 *						'font-size': '1.2em'
+	 *					}
+	 *				},
+	 *				small: {
+	 *					name: 'span',
+	 *					style: {
+	 *						'font-size': '0.8em'
+	 *					}
+	 *				}
+	 *			},
+	 *			upcastAlso: {
+	 *				big: viewElement => {
+	 *					const fontSize = viewElement.getStyle( 'font-size' );
+	 *
+	 *					if ( !fontSize ) {
+	 *						return null;
+	 *					}
+	 *
+	 *					const match = fontSize.match( /(\d+)\s*px/ );
+	 *
+	 *					if ( !match ) {
+	 *						return null;
+	 *					}
+	 *
+	 *					const size = Number( match[ 1 ] );
+	 *
+	 *					if ( viewElement.is( 'span' ) && size > 10 ) {
+	 *						return { name: true, style: [ 'font-size' ] };
+	 *					}
+	 *
+	 *					return null;
+	 *				},
+	 *				small: viewElement => {
+	 *					const fontSize = viewElement.getStyle( 'font-size' );
+	 *
+	 *					if ( !fontSize ) {
+	 *						return null;
+	 *					}
+	 *
+	 *					const match = fontSize.match( /(\d+)\s*px/ );
+	 *
+	 *					if ( !match ) {
+	 *						return null;
+	 *					}
+	 *
+	 *					const size = Number( match[ 1 ] );
+	 *
+	 *					if ( viewElement.is( 'span' ) && size < 10 ) {
+	 *						return { name: true, style: [ 'font-size' ] };
+	 *					}
+	 *
+	 *					return null;
+	 *				}
+	 *			}
+	 *		} );
+	 *
+	 * @param {~ConverterDefinition} definition Converter definition.
+	 */
+	attributeToElement( definition ) {
+		// Set up downcast converter.
+		this.for( 'downcast' ).add( downcastAttributeToElement( definition ) );
+
+		// Set up upcast converter.
+		for ( const { model, view } of _getAllUpcastDefinitions( definition ) ) {
+			this.for( 'upcast' ).add(
+				upcastElementToAttribute( {
+					view,
+					model,
+					priority: definition.priority
+				} )
+			);
+		}
+	}
+
+	/**
+	 * Sets up converters between the model and the view which convert a model attribute to a view attribute (and vice versa).
+	 * For example, `<image src='foo.jpg'></image>` is converted to `<img src='foo.jpg'></img>` (same attribute key and value).
+	 *
+	 * `definition.model` parameter specifies what model attribute should be converted from/to.
+	 * It can be a `{ key, values, [ name ] }` object or a `String`, which will be treated like `{ key: definition.model }`.
```

Comment:
`{ key, [ values ], [ name ] }` (value is optional too)

Reply:
As we are at it, I wanted to ask you, whether it is better to name that property `values` or `value` (it is always an array).